pkgdb/src/util.cc

```diff
@@ -326,6 +326,58 @@ displayableGlobbedPath( const flox::AttrPathGlob & attrs )
 
 /* -------------------------------------------------------------------------- */
 
+
+std::string
+joinWithDelim( const std::vector<std::string> & strings,
+               const std::string &              delim )
+{
+  auto fold
+    = [&]( std::string acc, std::string elem ) { return acc + delim + elem; };
+  std::string joined = std::accumulate( std::next( strings.begin() ),
+                                        strings.end(),
+                                        strings[0],
+                                        fold );
+  return joined;
+}
+
+/* -------------------------------------------------------------------------- */
+
+void
+printLog( const nix::Verbosity & lvl, const std::string & msg )
+{
+  if ( lvl <= nix::verbosity ) { nix::logger->log( lvl, msg ); }
+}
+
+void
```

Comment:
non-blocking recommendations: small 1 line functions like this can go in headers as :

```c++
// @file include/flox/core/util.hh


static inline void
printLog( const nix::Verbosity & lvl, const std::string & msg )
{
  if ( lvl <= nix::verbosity ) { nix::logger->log( lvl, msg ); }
}

static inline void
debugLog( const std::string & msg )
{
  printLog( nix::Verbosity::lvlDebug, msg );
}

// ...<SNIP>...
```

Reply:
I had them in a header and I ran into issues with duplicate symbols that I didn't know how to fix, so I went this route instead which somehow solved the problem.